### ISSUE TITLE
go client use Ctype_int as type _C type_ulong

### ISSUE THREAD

**Issue by chrislearn** [apache/pulsar#4095]
**Describe the bug**
When I use pulsar go client in ubuntu (windows subsystem), I install cpp client 2.3.1 and use go client tagged v2.3.1, but I got an error:
`
/home/chris/.gvm/pkgsets/go1.12.1/global/pkg/mod/github.com/kenorld/pulsar-client-go@v0.0.0-20190421093301-10f7ec0db253/pulsar/c_message.go:91:63: cannot use size (type _Ctype_int) as type _C
type_ulong in assignment
`

**To Reproduce**
1. download and install cpp client from official site.
2. use go client in pulsar repository in tag 2.3.1
3. run my app
`
/home/chris/.gvm/pkgsets/go1.12.1/global/pkg/mod/github.com/kenorld/pulsar-client-go@v0.0.0-20190421093301-10f7ec0db253/pulsar/c_message.go:91:63: cannot use size (type _Ctype_int) as type _C
type_ulong in assignment
`

**Screenshots**
![image](https://user-images.githubusercontent.com/5874864/56470286-266bde00-6477-11e9-9ae1-b9536890d27d.png)

**Desktop (please complete the following information):**
 - OS: Ubuntu installed in Windows 10 subsystem

**Comment by wolfstudy** [comment]
Hello, this problem seems to be that `pulsar-client-cpp` is not built successfully. Currently `pulsar-client-go` runs on `pulsar-client-cpp`, you can try the following command:

```
git clone git@github.com:apache/pulsar.git
cd apache/pulsar/pulsar-client-cpp
cmake .
make 
make install
```

**Comment by skyrocknroll** [comment]
We are facing the same in MacOS

**Comment by skyrocknroll** [comment]
@merlimat This commit 44c8fd552e1b005d8d531a2f9fb7200682794c46 should be commited into branch-2.3 to solve the issue ?

**Comment by juan-daishogroup** [comment]
We are experiencing the same issue in MacOS. Any ideas when this will be ready?

**Comment by skyrocknroll** [comment]
You can import "http://github.com/skyrocknroll/pulsar/pulsar-client-go/pulsar"

**Comment by juan-daishogroup** [comment]
Awesome! Thanks

**Comment by sijie** [comment]
Fixed in #4212
